diff --git a/src/iotjs.c b/src/iotjs.c
index 2268425..59e97f4 100644
--- a/src/iotjs.c
+++ b/src/iotjs.c
@@ -28,6 +28,7 @@
 #include "jerryscript.h"
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 
 
@@ -183,8 +184,11 @@ static void iotjs_uv_walk_to_close_callback(uv_handle_t* handle, void* arg) {
   iotjs_handlewrap_close(handle_wrap, NULL);
 }
 
+static int argc;
+static char **argv;
 
-int iotjs_entry(int argc, char** argv) {
+static int __attribute__ ((noinline))
+iotjs_entry_main(void) {
   // Initialize debug print.
   init_debug_settings();
 
@@ -233,3 +237,52 @@ terminate:
 
   return ret_code;
 }
+
+#define PATTERN (0xfe)
+#define STACK_MEASURE_RANGE (8 * 1024 * 1024)
+
+uint8_t stack[STACK_MEASURE_RANGE] __attribute__((aligned(16)));
+
+static void
+end_stack_measurement(void)
+{
+  uint8_t *stack_p = stack;
+  uint8_t *stack_end_p = stack + STACK_MEASURE_RANGE;
+
+  while(stack_p < stack_end_p)
+  {
+    if(*stack_p != PATTERN)
+    {
+      break;
+    }
+    stack_p++;
+  }
+
+  printf("Stack usage: %d\n", (int) (stack_end_p - stack_p));
+  exit(0);
+}
+
+int __attribute__ ((noinline))
+iotjs_entry(int argc_, char **argv_)
+{
+  uint8_t *stack_p;
+  uint8_t *stack_end_p;
+
+  argc = argc_;
+  argv = argv_;
+
+  stack_p = stack;
+  stack_end_p = stack + STACK_MEASURE_RANGE;
+
+  while(stack_p < stack_end_p)
+  {
+    *stack_p = PATTERN;
+    stack_p++;
+  }
+
+  __asm volatile ("mov sp, %0" : : "r" (stack_end_p));
+
+  iotjs_entry_main();
+  end_stack_measurement();
+  return 0;
+}
