diff --git a/apps/system/init/init.c b/apps/system/init/init.c
index cc4c540..c545503 100644
--- a/apps/system/init/init.c
+++ b/apps/system/init/init.c
@@ -130,6 +130,10 @@ int preapp_start(int argc, char *argv[])
 	}
 #endif

+#ifdef CONFIG_FS_ROMFS
+	mount("/dev/smart4rom9", "/rom", "romfs", 0, NULL);
+#endif
+
 #if defined(CONFIG_LIB_USRWORK) || defined(CONFIG_TASH)
 error_out:
 	return pid;
diff --git a/apps/system/jerryscript/jerry_main.c b/apps/system/jerryscript/jerry_main.c
index 9ea5202..3c715ef 100644
--- a/apps/system/jerryscript/jerry_main.c
+++ b/apps/system/jerryscript/jerry_main.c
@@ -452,6 +452,7 @@ jerry_cmd_main (int argc, char *argv[])
   jerry_release_value (ret_value);
   jerry_cleanup ();

+  printf("RESULT: %d\n", ret_code);
   return ret_code;
 } /* jerry_cmd_main */

diff --git a/build/configs/artik053/artik053_download.sh b/build/configs/artik053/artik053_download.sh
index 711d131..7f8eee0 100755
--- a/build/configs/artik053/artik053_download.sh
+++ b/build/configs/artik053/artik053_download.sh
@@ -72,6 +72,7 @@ main()
 				flash_write sssfw ../../bin/sssfw.bin; 		\
 				flash_write wlanfw ../../bin/wlanfw.bin;	\
 				flash_write os ../../../../output/bin/tinyara_head.bin;	\
+				flash_write rom ../../../../output/bin/rom.img;	\
 			exit'
 			popd
 			;;
diff --git a/build/configs/artik053/jerryscript/defconfig b/build/configs/artik053/jerryscript/defconfig
index a24bdfd..b88ba62 100644
--- a/build/configs/artik053/jerryscript/defconfig
+++ b/build/configs/artik053/jerryscript/defconfig
@@ -281,7 +281,7 @@ CONFIG_ARTIK053_FLASH_CAPACITY=8388608
 CONFIG_ARTIK053_FLASH_PAGE_SIZE=4096
 CONFIG_ARTIK053_FLASH_PART=y
 CONFIG_ARTIK053_FLASH_MINOR=0
-CONFIG_ARTIK053_FLASH_PART_LIST="16,48,192,32,512,2400,1536,1536,1000,400,8,512,"
+CONFIG_ARTIK053_FLASH_PART_LIST="16,48,192,32,512,2400,1536,1536,800,600,8,512,"
 CONFIG_ARTIK053_FLASH_PART_TYPE="none,ftl,none,none,none,none,none,ftl,smartfs,romfs,config,none,"
 CONFIG_ARTIK053_FLASH_PART_NAME="bl1,sssro,bl2,sssfw,wlanfw,os,factory,ota,user,rom,nvram,sssrw,"
 CONFIG_ARTIK053_AUTOMOUNT=y
@@ -884,4 +884,4 @@ CONFIG_SYSTEM_VI_DEBUGLEVEL=0
 #
 # wpa_supplicant
 #
-# CONFIG_WPA_SUPPLICANT is not set
\ No newline at end of file
+# CONFIG_WPA_SUPPLICANT is not set
diff --git a/build/configs/artik053/tools/openocd/artik053.cfg b/build/configs/artik053/tools/openocd/artik053.cfg
index ebba174..6871716 100644
--- a/build/configs/artik053/tools/openocd/artik053.cfg
+++ b/build/configs/artik053/tools/openocd/artik053.cfg
@@ -253,3 +253,10 @@ proc flash_erase_part { partition } {
 }
 add_usage_text flash_erase_part "{partition|all}"
 add_help_text flash_erase_part "Erase a partition. If specified 'all', all r/w partitions are erased."
+
+proc reset {} {
+	prepare_target
+	flash_enable_write
+	catch {mwb 0x80090400 0x1}
+	exit 0
+}
diff --git a/build/configs/artik053/tools/openocd/partition_map.cfg b/build/configs/artik053/tools/openocd/partition_map.cfg
index 10455a4..740f31c 100644
--- a/build/configs/artik053/tools/openocd/partition_map.cfg
+++ b/build/configs/artik053/tools/openocd/partition_map.cfg
@@ -11,7 +11,8 @@ set partition_list {
 	os	{ "OS"			0x040C8000	0x00258000	0 }
 	factory	{ "Factory Reset"	0x04320000	0x00180000	0 }
 	ota	{ "OTA download"	0x044A0000	0x00180000	0 }
-	user	{ "USER R/W"		0x04620000	0x0015E000	0 }
+	user	{ "USER R/W"		0x04620000	0x000C8000	0 }
+	rom	{ "ROM FS"		0x046E8000	0x00096000	0 }
 	nvram	{ "WiFi NVRAM"		0x0477E000	0x00002000	1 }
 	sssrw	{ "SSS R/W Key"		0x04780000	0x00080000	1 }
 }
