diff --git a/build/configs/artik053/artik053_download.sh b/build/configs/artik053/artik053_download.sh
index eabbc3f..4e40e1a 100755
--- a/build/configs/artik053/artik053_download.sh
+++ b/build/configs/artik053/artik053_download.sh
@@ -129,6 +129,15 @@ main()
 			popd
 			;;
 
+		RESET|reset)
+			echo "board reset"
+
+			pushd ${OPENOCD_DIR_PATH}
+			${OPENOCD_BIN_PATH}/openocd -f artik053.cfg -c ' 	\
+			reset; \
+			exit'
+			popd
+			;;
 		*)
 			echo "${arg} is not suppported in artik053"
 			echo "Usage : make download [ ALL | ERASE_USERFS ]"
diff --git a/build/configs/artik053/tools/openocd/artik053.cfg b/build/configs/artik053/tools/openocd/artik053.cfg
index 1e18d50..bddfb64 100644
--- a/build/configs/artik053/tools/openocd/artik053.cfg
+++ b/build/configs/artik053/tools/openocd/artik053.cfg
@@ -251,5 +251,13 @@ proc flash_erase_part { partition } {
 		flash_erase $start $size -silent
 	}
 }
+
+proc reset {} {
+	prepare_target
+	flash_enable_write
+	catch {mwb 0x80090400 0x1}
+	exit 0
+}
+
 add_usage_text flash_erase_part "{partition|all}"
 add_help_text flash_erase_part "Erase a partition. If specified 'all', all r/w partitions are erased."
