diff --git a/jerry-core/parser/js/js-parser-statm.c b/jerry-core/parser/js/js-parser-statm.c
index 4330ef4..26f8899 100644
--- a/jerry-core/parser/js/js-parser-statm.c
+++ b/jerry-core/parser/js/js-parser-statm.c
@@ -332,7 +332,7 @@ parser_parse_var_statement (parser_context_t *context_p) /**< context */
       if ((JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_CONNECTED)
           && ident_line_counter != context_p->last_breakpoint_line)
       {
-        parser_emit_cbc (context_p, CBC_BREAKPOINT_DISABLED);
+        parser_emit_cbc (context_p, CBC_BREAKPOINT_ENABLED);
         parser_flush_cbc (context_p);
 
         parser_append_breakpoint_info (context_p, JERRY_DEBUGGER_BREAKPOINT_LIST, ident_line_counter);
@@ -1751,7 +1751,7 @@ parser_parse_statements (parser_context_t *context_p) /**< context */
       {
         JERRY_ASSERT (context_p->last_breakpoint_line == 0);
 
-        parser_emit_cbc (context_p, CBC_BREAKPOINT_DISABLED);
+        parser_emit_cbc (context_p, CBC_BREAKPOINT_ENABLED);
         parser_flush_cbc (context_p);
 
         parser_append_breakpoint_info (context_p, JERRY_DEBUGGER_BREAKPOINT_LIST, context_p->token.line);
@@ -1821,7 +1821,7 @@ parser_parse_statements (parser_context_t *context_p) /**< context */
         && context_p->token.type != LEXER_KEYW_CASE
         && context_p->token.type != LEXER_KEYW_DEFAULT)
     {
-      parser_emit_cbc (context_p, CBC_BREAKPOINT_DISABLED);
+      parser_emit_cbc (context_p, CBC_BREAKPOINT_ENABLED);
       parser_flush_cbc (context_p);
 
       parser_append_breakpoint_info (context_p, JERRY_DEBUGGER_BREAKPOINT_LIST, context_p->token.line);
diff --git a/jerry-core/parser/js/js-parser.c b/jerry-core/parser/js/js-parser.c
index 4f145e3..707ca01 100644
--- a/jerry-core/parser/js/js-parser.c
+++ b/jerry-core/parser/js/js-parser.c
@@ -1492,7 +1492,7 @@ parser_post_processing (parser_context_t *context_p) /**< context */
       && !(context_p->status_flags & PARSER_DEBUGGER_BREAKPOINT_APPENDED))
   {
     /* Always provide at least one breakpoint. */
-    parser_emit_cbc (context_p, CBC_BREAKPOINT_DISABLED);
+    parser_emit_cbc (context_p, CBC_BREAKPOINT_ENABLED);
     parser_flush_cbc (context_p);
 
     parser_append_breakpoint_info (context_p, JERRY_DEBUGGER_BREAKPOINT_LIST, context_p->token.line);
@@ -1887,7 +1887,7 @@ parser_post_processing (parser_context_t *context_p) /**< context */
     flags = cbc_flags[opcode];
 
 #ifdef JERRY_DEBUGGER
-    if (opcode == CBC_BREAKPOINT_DISABLED)
+    if (opcode == CBC_BREAKPOINT_ENABLED)
     {
       uint32_t bp_offset = (uint32_t) (((uint8_t *) dst_p) - ((uint8_t *) compiled_code_p) - 1);
       parser_append_breakpoint_info (context_p, JERRY_DEBUGGER_BREAKPOINT_OFFSET_LIST, bp_offset);
