diff --git a/src/unix/fs.c b/src/unix/fs.c
index 5971915..4026d14 100644
--- a/src/unix/fs.c
+++ b/src/unix/fs.c
@@ -98,7 +98,7 @@
     if (cb == NULL) {                                                         \
       req->path = path;                                                       \
     } else {                                                                  \
-      req->path = strdup(path);                                               \
+      req->path = uv__strdup(path);                                           \
       if (req->path == NULL) {                                                \
         uv__req_unregister(loop, req);                                        \
         return -ENOMEM;                                                       \
diff --git a/src/uv-common.c b/src/uv-common.c
index 5c12d82..7e0cb57 100644
--- a/src/uv-common.c
+++ b/src/uv-common.c
@@ -60,14 +60,18 @@ typedef struct {
   uv_free_func local_free;
 } uv__allocator_t;
 
+void jstest_free(void*);
+void* jstest_malloc(size_t);
+void* jstest_calloc(size_t, size_t);
+void* jstest_realloc(void*, size_t);
+
 static uv__allocator_t uv__allocator = {
-  malloc,
-  realloc,
-  calloc,
-  free,
+  jstest_malloc,
+  jstest_realloc,
+  jstest_calloc,
+  jstest_free,
 };
 
-#if defined(__APPLE__) || defined(_WIN32) || defined(TUV_FEATURE_PIPE)
 char* uv__strdup(const char* s) {
   size_t len = strlen(s) + 1;
   char* m = uv__malloc(len);
@@ -75,7 +79,6 @@ char* uv__strdup(const char* s) {
     return NULL;
   return memcpy(m, s, len);
 }
-#endif
 
 void* uv__malloc(size_t size) {
   return uv__allocator.local_malloc(size);
diff --git a/src/uv-common.h b/src/uv-common.h
index e52b529..a24de69 100644
--- a/src/uv-common.h
+++ b/src/uv-common.h
@@ -239,9 +239,7 @@ void uv__fs_scandir_cleanup(uv_fs_t* req);
 
 /* Allocator prototypes */
 void *uv__calloc(size_t count, size_t size);
-#if defined(__APPLE__) || defined(_WIN32) || defined(TUV_FEATURE_PIPE)
 char *uv__strdup(const char* s);
-#endif
 void* uv__malloc(size_t size);
 void uv__free(void* ptr);
 void* uv__realloc(void* ptr, size_t size);
