diff --git a/os/FlatLibs.mk b/os/FlatLibs.mk
index de0ec9c..65211e3 100644
--- a/os/FlatLibs.mk
+++ b/os/FlatLibs.mk
@@ -153,6 +153,7 @@ TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)libiotjs$(LIBEXT)
 TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)libjerry-core$(LIBEXT)
 TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)libtuv$(LIBEXT)
 TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)libjerry-libm$(LIBEXT)
+TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)libjerry-port-default$(LIBEXT)
 endif
 
 # Add library for external bcm support.
diff --git a/os/LibTargets.mk b/os/LibTargets.mk
index 95be375..3a41a1f 100644
--- a/os/LibTargets.mk
+++ b/os/LibTargets.mk
@@ -242,6 +242,9 @@ $(LIBRARIES_DIR)$(DELIM)libtuv$(LIBEXT): $(IOTJS_LIB_DIR)$(DELIM)libtuv$(LIBEXT)
 
 $(LIBRARIES_DIR)$(DELIM)libjerry-libm$(LIBEXT): $(IOTJS_LIB_DIR)$(DELIM)libjerry-libm$(LIBEXT)
 	$(Q) install $(IOTJS_LIB_DIR)$(DELIM)libjerry-libm$(LIBEXT) $(LIBRARIES_DIR)$(DELIM)libjerry-libm$(LIBEXT)
+
+$(LIBRARIES_DIR)$(DELIM)libjerry-port-default$(LIBEXT): $(IOTJS_LIB_DIR)$(DELIM)libjerry-port-default$(LIBEXT)
+	$(Q) install $(IOTJS_LIB_DIR)$(DELIM)libjerry-port-default$(LIBEXT) $(LIBRARIES_DIR)$(DELIM)libjerry-port-default$(LIBEXT)
 endif
 
 # Possible non-kernel builds
diff --git a/os/ProtectedLibs.mk b/os/ProtectedLibs.mk
index 42b5fd8..62bbb58 100644
--- a/os/ProtectedLibs.mk
+++ b/os/ProtectedLibs.mk
@@ -148,6 +148,7 @@ USERLIBS += $(LIBRARIES_DIR)$(DELIM)libiotjs$(LIBEXT)
 USERLIBS += $(LIBRARIES_DIR)$(DELIM)libjerry-core$(LIBEXT)
 USERLIBS += $(LIBRARIES_DIR)$(DELIM)libtuv$(LIBEXT)
 USERLIBS += $(LIBRARIES_DIR)$(DELIM)libjerry-libm$(LIBEXT)
+USERLIBS += $(LIBRARIES_DIR)$(DELIM)libjerry-port-default$(LIBEXT)
 endif
 
 EXPORTLIBS = $(USERLIBS)
