diff --git a/src/platform/tizenrt/iotjs_main_tizenrt.c b/src/platform/tizenrt/iotjs_main_tizenrt.c
index 00b7b58..8bd9415 100644
--- a/src/platform/tizenrt/iotjs_main_tizenrt.c
+++ b/src/platform/tizenrt/iotjs_main_tizenrt.c
@@ -90,12 +90,9 @@ struct iotjs_thread_arg {
 pthread_addr_t iotjs_thread(void *thread_arg) {
   struct iotjs_thread_arg *arg = thread_arg;
 
-#ifdef CONFIG_DEBUG_VERBOSE
   int ret = iotjs_entry(arg->argc, arg->argv);
   printf("IoT.js Result: %d\n", ret);
-#else
-  iotjs_entry(arg->argc, arg->argv);
-#endif
+
   tuv_cleanup();
 
   sleep(1);
