diff --git a/jerry-core/jmem/jmem-heap.c b/jerry-core/jmem/jmem-heap.c
index f15ba90..680cc3a 100644
--- a/jerry-core/jmem/jmem-heap.c
+++ b/jerry-core/jmem/jmem-heap.c
@@ -580,7 +580,7 @@ jmem_heap_stats_print (void)
 {
   jmem_heap_stats_t *heap_stats = &JERRY_CONTEXT (jmem_heap_stats);
 
-  JERRY_DEBUG_MSG ("Heap stats:\n"
+  printf ("Heap stats:\n"
                    "  Heap size = %zu bytes\n"
                    "  Allocated = %zu bytes\n"
                    "  Peak allocated = %zu bytes\n"
@@ -608,7 +608,7 @@ jmem_heap_stats_print (void)
                    heap_stats->property_bytes,
                    heap_stats->peak_property_bytes);
 #ifndef JERRY_SYSTEM_ALLOCATOR
-  JERRY_DEBUG_MSG ("  Skip-ahead ratio = %zu.%04zu\n"
+  printf ("  Skip-ahead ratio = %zu.%04zu\n"
                    "  Average alloc iteration = %zu.%04zu\n"
                    "  Average free iteration = %zu.%04zu\n",
                    heap_stats->skip_count / heap_stats->nonskip_count,
