diff --git a/build/configs/artik05x/artik05x_download.sh b/build/configs/artik05x/artik05x_download.sh
index cea0f64..589f6ec 100755
--- a/build/configs/artik05x/artik05x_download.sh
+++ b/build/configs/artik05x/artik05x_download.sh
@@ -246,6 +246,12 @@ while test $# -gt 0; do
 		ALL|OS|ROMFS|BL1|BL2|SSSFW|WLANFW|OTA|all|os|romfs|bl1|bl2|sssfw|wlanfw|ota)
 			download $1
 			;;
+		RESET|reset)
+			echo "board reset"
+			pushd ${OPENOCD_DIR_PATH} > /dev/null
+			${OPENOCD} -f ${CFG_FILE} -s ${SCRIPTS_PATH} -c 'reset; exit'
+			popd
+			;;
 		ERASE_*)
 			erase
 			;;
diff --git a/build/configs/artik05x/scripts/artik05x.cfg b/build/configs/artik05x/scripts/artik05x.cfg
index 8da4c4d..56f3006 100644
--- a/build/configs/artik05x/scripts/artik05x.cfg
+++ b/build/configs/artik05x/scripts/artik05x.cfg
@@ -101,6 +101,13 @@ proc flash_poll {} {
 	}
 }
 
+proc reset {} {
+	prepare_target
+	flash_enable_write
+	catch {mwb 0x80090400 0x1}
+	exit 0
+}
+
 proc prepare_target {} {
 	init
 	catch { reset halt }
